### PR TITLE
Updates to bindings and docs

### PR DESCRIPTION
**Pull request by urschrei**
- specify `proj.4`, now that `proj.5` has been released
- Update dependencies
- Replace use of tuples (not layout-safe for FFI) with `c_void`
- Add rudimentary inline docs
- Add example from tests to README

### CODE REVIEW

**Comment by urschrei** [comment]
(preliminary steps toward addressing some issues in https://github.com/georust/rust-geo/issues/32)

File: README.md [hunk 1]
```diff
@@ -1,3 +1,22 @@
 # rust-proj
 
-Rust bindings for [proj](https://github.com/OSGeo/proj.4)
+Rust bindings for [proj.4](https://github.com/OSGeo/proj.4)
+
+# Example
```
Comment (frewsxcv) [inline README.md]: ❤️

File: README.md [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+use geo::types::Point;
+
+# reproject coordinates from Stereo70 with custom params into WGS84 lon and lat coordinates
+let wgs84_name = "+proj=longlat +datum=WGS84 +no_defs";
+let wgs84 = Proj::new(wgs84_name).unwrap();
+let stereo70 = Proj::new(
+    "+proj=sterea +lat_0=46 +lon_0=25 +k=0.99975 +x_0=500000 +y_0=500000 +ellps=krass +units=m +no_defs"
+    ).unwrap();
+let rp = stereo70.project(&wgs84, Point::new(500000., 500000.));
+# New Point coords are (0.436332, 0.802851)
```
Comment (frewsxcv) [inline README.md]: alternatively, you could do an `assert_eq!`

File: README.md [hunk 3]
```diff
@@ -1,3 +1,22 @@
...
+# Example
+## Reproject from [Stereo70](https://epsg.io/3844) to [WGS84](https://epsg.io/4326)
+```rust
+extern crate proj;
+use proj::Proj;
+
+extern crate geo;
+use geo::types::Point;
+
+# reproject coordinates from Stereo70 with custom params into WGS84 lon and lat coordinates
```
Comment (frewsxcv) [inline README.md]: comments in rust start with `//` 😅

Comment (urschrei) [reply README.md]: 🤭

**Comment by frewsxcv** [comment]
looks like i need to give publish permissions, one sec

**Comment by frewsxcv** [comment]
done

**Comment by urschrei** [comment]
<3 I think I've set them up correctly for georust core for [proj-sys](https://github.com/georust/proj-sys)

**Comment by frewsxcv** [comment]
yep, looks good! also gonna close https://github.com/georust/rust-proj/issues/1

**Pull request merged**